packages/app/components/SelectNetwork.tsx

```diff
@@ -1,29 +1,85 @@
 "use client";
 
-import { Button, Icon } from "@/ui";
-import { useAccount, useNetwork, useSwitchNetwork } from "wagmi";
+import { Fragment, useEffect, useState } from "react";
+
+import type { Chain } from "viem/chains";
 import { ChainIcon } from "connectkit";
 import { Listbox, Transition } from "@headlessui/react";
-import { Fragment } from "react";
-import { parseAsInteger, useQueryState } from "next-usequerystate";
+import { useAccount, useNetwork, useSwitchNetwork } from "wagmi";
+
+import { Button, Icon } from "@/ui";
+import { config } from "@/providers";
+import { useStackboxFormContext } from "@/contexts";
+
+interface WagmiChain extends Chain {
+  unsupported?: boolean;
+}
 
 export const SelectNetwork = () => {
-  const [, setChainId] = useQueryState("chainId", parseAsInteger);
+  const { stackboxFormState } = useStackboxFormContext();
+  const [, setChainId] = stackboxFormState.chainIdState;
   const { switchNetwork } = useSwitchNetwork({
     onSuccess(data) {
       setChainId(data.id);
     },
   });
-  const { chain, chains } = useNetwork();
+  const { chain } = useNetwork();
   const { isConnected } = useAccount();
+  const [selectedChain, setSelectedChain] = useState<WagmiChain>();
+  const [allowedChains, setAllowedChains] = useState<WagmiChain[] | undefined>(
+    []
+  );
+
+  const handleOfflineNetworkSwitch = (newChainId: number) => {
+    config.setState((oldState: any) => {
+      const { publicClient } = oldState;
+      const { chains } = publicClient;
+
+      const offlineChain = chains?.find(
```

Comment:
Same here, IMO offline doesn't make sense.
In this case we're selecting a new chain if it's on allowedChain.

```suggestion
      const newChain = chains?.find(
```

Reply:
What if the `newChain` is not on allowed chains? Don't know if this can happen, but if take url params, maybe it can.

Reply:
This value in specific happens `onChange` in the switch listbox, and the switch listbox only displays the allowed chains (which are the ones set in WAGMI config).

The issue you mention can happen on the Stackbox form state, will check it right now :D